### PR TITLE
Add --components and --architectures to remote command

### LINKED ISSUE #4: Cannot specify components or architectures
```
 pulp deb remote create --help
Usage: pulp deb remote create [OPTIONS]

  Create a remote.

Options:
  --name TEXT                     [required]
  --url TEXT                      [required]
  --ca-cert TEXT                  a PEM encoded CA certificate
  --client-cert TEXT              a PEM encoded client certificate
  --client-key TEXT               a PEM encode private key
  --connect-timeout FLOAT
  --download-concurrency INTEGER  total number of simultaneous connections
  --password TEXT
  --proxy-url TEXT
  --proxy-username TEXT
  --proxy-password TEXT
  --rate-limit INTEGER            limit download rate in requests per second
  --sock-connect-timeout FLOAT
  --sock-read-timeout FLOAT
  --tls-validation BOOLEAN
  --total-timeout FLOAT
  --username TEXT
  --policy [immediate|on_demand|streamed]
  --distributions TEXT            [required]
  --help                          Show this message and exit.

```

No support for components or architectures seems to exist

### PR DESCRIPTION
**Pull request by quba42**
closes #4

### CODE REVIEW

File: pulpcore/cli/deb/remote.py [hunk 1]
```diff
@@ -44,6 +44,8 @@ def remote(ctx: click.Context, pulp_ctx: PulpContext, remote_type: str) -> None:
         "--policy", type=click.Choice(["immediate", "on_demand", "streamed"], case_sensitive=False)
     ),
     click.option("--distributions", required=True),
+    click.option("--components", help=_("Whitespace separated list of components.")),
```
Comment (mdellweg) [inline pulpcore/cli/deb/remote.py:47]: We usually use something like this:
```suggestion
    click.option("--component", "components", multiple=True, help=_("Whitespace separated list of components.")),
```

You'd specify all the components separately that way.

**Review by mdellweg — APPROVED** [review]

File: pulpcore/cli/deb/context.py [hunk 2]
```diff
@@ -75,6 +75,7 @@ class PulpAptRemoteContext(PulpEntityContext):
     CREATE_ID = "remotes_deb_apt_create"
     UPDATE_ID = "remotes_deb_apt_partial_update"
     DELETE_ID = "remotes_deb_apt_delete"
+    NULLABLES = {"architecture", "component"}
```
Comment (mdellweg) [inline pulpcore/cli/deb/context.py:78]:
```suggestion
    NULLABLES = {"architectures", "components"}
```

Comment (quba42) [reply pulpcore/cli/deb/context.py]: Fixed.

File: pulpcore/cli/deb/context.py [hunk 3]
```diff
@@ -75,12 +75,25 @@ class PulpAptRemoteContext(PulpEntityContext):
...
+    def tuple_to_whitespace_separated_string(field_name: str, body: EntityDefinition) -> None:
+        """
+        Adjusts the field field_name within the body in place.
+        Tuple-lists are turned into whitespace separated strings.
+        """
+        field = body.pop(field_name, None)
+        if field is not None:
+            if not isinstance(field, tuple):
+                # Is this exception condition needed?! I don't think it can not be a tuple...
+                raise Exception
```
Comment (quba42) [inline pulpcore/cli/deb/context.py:89]: Like the comment says, I am not sure if this is really needed. Will perform some targeted testing.

File: pulpcore/cli/deb/remote.py (hunk 1 above)
Comment (quba42) [reply pulpcore/cli/deb/remote.py]: Latest implementation uses multiple=True.

File: pulpcore/cli/deb/context.py (hunk 3 above)
Comment (mdellweg) [reply pulpcore/cli/deb/context.py:90]: You should be careful here. Despite distributions, components and architectures may be `None` deliberately to clear them on the server.

**Comment by quba42** [comment]
Using bash instead of sh makes the tests run as intended...

**Review by mdellweg — APPROVED** [review]

**Pull request merged**